core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala

```diff
@@ -440,6 +446,9 @@ private[spark] class MesosClusterScheduler(
         .mkString(",")
       options ++= Seq("--py-files", formattedFiles)
     }
+    desc.schedulerProperties
+      .filter { case (key, _) => !replicatedOptionsBlacklist.contains(key) }
+      .foreach { case (key, value) => options ++= Seq("--conf", s"""$key="$value"""") }
```

Comment:
That's a good point, `CommandInfo` is using `/bin/sh` to launch the command. :confused: 

Spaces should be ok, everything else won't be correctly escaped. Skimming through Spark properties I think the only ones that could pose problems are `spark.authenticate.secret` and the other passwords (SSL, etc.). Still, this needs a solution.